File: app/src/main/java/com/example/rafaelanastacioalves/moby/entitydetailing/LiveDataEntityDetailsViewModel.java
package com.example.rafaelanastacioalves.moby.entitydetailing;

import android.arch.lifecycle.LiveData;
import android.arch.lifecycle.ViewModel;

import com.example.rafaelanastacioalves.moby.domain.entities.EntityDetails;
import com.example.rafaelanastacioalves.moby.domain.entities.Resource;
import com.example.rafaelanastacioalves.moby.domain.interactors.EntityDetailingInteractor;


public class LiveDataEntityDetailsViewModel extends ViewModel {

    private EntityDetailingInteractor mEntityDetailInteractor;
    private LiveData<Resource<EntityDetails>> mEntityDetails;

    public LiveDataEntityDetailsViewModel(EntityDetailingInteractor mInteractor) {
        this.mEntityDetailInteractor = mInteractor;
    }

    public LiveData<Resource<EntityDetails>> getEntityDetails(String tripPackageId) {
        EntityDetailingInteractor.RequestValues requestValues = new EntityDetailingInteractor.RequestValues(tripPackageId);
        mEntityDetails = mEntityDetailInteractor.execute(requestValues);
        return mEntityDetails;
    }


}

File: app/src/main/java/com/example/rafaelanastacioalves/moby/entitydetailing/LiveDataEntityDetailsViewModel.kt
package com.example.rafaelanastacioalves.moby.entitydetailing

import android.arch.lifecycle.MutableLiveData
import com.example.rafaelanastacioalves.moby.base.BaseViewModel

import com.example.rafaelanastacioalves.moby.entities.EntityDetails
import com.example.rafaelanastacioalves.moby.retrofit.APIClient

import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response
import timber.log.Timber
import javax.inject.Inject

class LiveDataEntityDetailsViewModel : BaseViewModel() {

    val entityDetails = MutableLiveData<EntityDetails>()

    @Inject
    lateinit var apiClient: APIClient


    fun loadData(tripPackageId: String?) {
        Timber.i("LiveDataEntityDetailsViewModel loadData")

        if (entityDetails.value != null) {
            return
        }


        if (tripPackageId == null) {
            Timber.w("loadInBackground - not supposed to have null variable here")
            return
        }
        val call = apiClient.getTripPackageDetails(tripPackageId)


        call.enqueue(object : Callback<EntityDetails> {
            override fun onResponse(call: Call<EntityDetails>, response: Response<EntityDetails>) {
                if (response.isSuccessful) {
                    Timber.i("response Successful")
                    entityDetails.postValue(response.body())

                } else {
                    Timber.e(response.message())
                }
            }

            override fun onFailure(call: Call<EntityDetails>, t: Throwable) {
                //TODO add more error management
                t.printStackTrace()
            }

        })


    }
}

